__tests__/Goals/components/Create-Goals-test.tsx

```diff
@@ -59,13 +65,57 @@ describe('MainScreen', () => {
     expect(userContainer).toBeNull();
   });
 
-  test.skip('navigates to FormScreen when "Create" button is pressed', () => {
-    const navigateMock = jest.fn();
-    const { getByText } = render(
-      <MembersPage navigation={{ push: navigateMock }} />,
+  test('navigates to Goal Screen when "Create Goal" button is pressed', async () => {
+    const spyAlert = jest.spyOn(Alert, 'alert');
+    axios.get.mockResolvedValue({
+      data: { users: mockUsersData, message: 'Users returned successfully!' },
+    });
+    axios.post.mockResolvedValue(mockGoalData);
+
+    const {
+      getByTestId,
+      getByPlaceholderText,
+      getByText,
+      findByTestId,
+      getAllByText,
+    } = render(<CreateGoals navigation={navigationProp} />);
+
+    const createGoalButton = getByText(/create goal/i);
+    const titleInput = getByPlaceholderText(
+      'Enter title max of 50 characters.',
     );
-    const createButton = getByText('Create');
-    fireEvent.press(createButton);
-    expect(navigateMock).toHaveBeenCalledWith('Form screen');
+    const descriptionInput = getByPlaceholderText('Enter max 200 characters.');
+    const selectUserButton = getByTestId('dropdown');
+
+    fireEvent.press(selectUserButton);
+    const userContainer = await findByTestId('user-container');
+    expect(userContainer).toBeTruthy();
+
+    const userInput = getByPlaceholderText('Search User');
+    await waitFor(() => {
+      fireEvent.changeText(userInput, 'test user');
+    });
+    const userItems = getAllByText(/test user/i);
+    expect(userItems).toBeTruthy();
+
+    await waitFor(() => {
+      fireEvent.press(userItems[0]);
+      fireEvent.changeText(titleInput, 'Test Goal');
+      fireEvent.changeText(descriptionInput, 'Test Description');
+    });
+
+    fireEvent.press(createGoalButton);
+
+    await waitFor(() => {
+      expect(Alert.alert).toHaveBeenCalledWith(
+        'Success',
+        `Task has been created and assigned to test user`,
+        [{ text: 'OK', onPress: expect.any(Function) }],
+      );
+    });
+
+    // @ts-ignore
```

Comment:
Why did you ignore the typescript check?

Reply:
I referred to this https://github.com/callstack/react-native-testing-library/issues/218#issuecomment-775905734

Reply:
Can you tell the error you are facing when removing this ts ignore command

Reply:
![image](https://github.com/user-attachments/assets/ada66975-9cdf-49c3-9bfd-61a58eabbb66)

Reply:
I can get rid of the ts error by asserting 
spyAlert.mock.calls?.[0]?.[2]?.[0].onPress!();

Reply:
Try to make it like this
[spyAlert.mock.calls?.[0]?.[2]?.[0]?.onPress](https://github.com/callstack/react-native-testing-library/issues/218#issuecomment-2117747532)

Reply:
![image](https://github.com/user-attachments/assets/53097e16-a94e-4d87-8a6c-5c01e99f0539)

Reply:
call it here like this onPress()

Reply:
do this then 
spyAlert.mock.calls?.[0]?.[2]?.[0].onPress!();

or the one mention in that link

Reply:
Yes and by using this does the tests are passing?

Reply:
Yes